lib/apiservers/engine/backends/archive.go

```diff
@@ -250,12 +250,9 @@ func NewArchiveStreamWriterMap(mounts []types.MountPoint, containerDestPath stri
 		// Here, we must tell the portlayer to remove "mnt/A".  The key to determining whether to
 		// strip "A" or "mnt/A" is based on the container destination path.
 		if containerDestPath != "/" && strings.HasPrefix(aw.mountPoint.Destination, containerDestPath) {
-			aw.filterSpec.StripPath = strings.TrimPrefix(aw.mountPoint.Destination, containerDestPath)
+			aw.filterSpec = vicarchive.GenerateFilterSpec(containerDestPath, aw.mountPoint, !strings.Contains(aw.mountPoint, containerDestPath), vicarchive.CopyTo)
```

Comment:
primaryTarget = !strings.Contains(aw.mountPoint, containerDestPath) is incorrect. It's more complicated than this.
Ie given mountpoints: 
```
/mnt/volA/
/mnt/volA/volC/
/mnt/VolB/
```
container destination: /mnt/volA/volC/fileABCD.txt.
The primary mount is /mnt/volA/volC. 
But your code will give
```
/mnt/volA/ - primary target = !strings.Contains(aw.mountPoint, containerDestPath) = false
/mnt/volA/volC/ - primary target = !strings.Contains(aw.mountPoint, containerDestPath) = false
/mnt/VolB/ - primary target = !strings.Contains(aw.mountPoint, containerDestPath) = true
```